cli/azd/pkg/infra/azure_resource_manager.go

```diff
@@ -38,17 +38,38 @@ func NewAzureResourceManager(azCli azcli.AzCli) *AzureResourceManager {
 	}
 }
 
+var hasTargetResourceData = func(operation *armresources.DeploymentOperation) bool {
```

Comment:
Just a stylistic comment - I'm not sure these functions are pulling their own weight.  My personal preference is that we would just remove these functions and inline the checks.  In each of these cases it feels like we're testing exactly one property of the object and just seeing the code at the place we do the check is more instructive.  I don't have to wonder what logic is behind these methods.

I do like how we used these names to break up a larger conditional into small pieces and document each step.  But I think the real value we got was from breaking up the conditional, we could inline the tests and the code would still read well (and you'd end up having to jump around less).

Perhaps another reason we introduced these changes was because we wanted to call these functions from multiple methods and felt bad about duplicating the code?  In the same way that go prefers "a little bit of code copying to taking a dependency", I feel like it also embraces a little bit of duplication in cases like this as well.

Reply:
Yes, it was mostly for self-documenting a big `if <condition>` where the condition has checking multiple things and each thing is a long line.  It was taking a few seconds to read it all and understand what the condition is doing.
So I split it so is really simple to read what the validation is doing.

I started by adding a comment to each validation line, like

```
if operation.Properties.TargetResource != nil &&  // Check tag
   *operation.Properties.TargetResource.ResourceType == string(AzureResourceTypeDeployment) && // is Deployment
    strings.TrimSpace(*operation.Properties.TargetResource.ResourceType) != "" { // has resource Type
...
}
```

Then I realized that the comment could actually become part of the code, instead of just a comment.
Code must be read, while comments are some times ignored..  

So I remembered something I read long time ago about making the code describe what it is doing instead of depending too much on comments.

I like how it is easier to read

```
if hasTargetResourceData(op) &&
   isDeploymentType(op) &&
   hasResourceType(op)
```

compared to previous longer if statement..  where each condition is quickly telling what it is checking... instead of the developer doing the `mental digest` to see what it is doing.

But I am fine returning to the longer `if` statements =/  ..